### PR TITLE
Process USB vendor requests in system thread

### PR DESCRIPTION
**Pull request by avtolstoy**
### Problem

Most of the currently supported vendor requests should be executed on system thread instead of being processed in ISR.

### Solution

Utilize `SystemISRTaskQueue` for vendor requests.

### Steps to Test

N/A

### Example App

N/A

### References

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: system/src/system_control.cpp [hunk 1]
```diff
@@ -181,10 +170,13 @@ uint8_t SystemControlInterface::handleVendorRequest(HAL_USB_SetupRequest* req) {
         } else {
           // Return as string
           String id = System.deviceID();
```
Comment (sergeuz) [inline system/src/system_control.cpp]: Unused variable

File: system/inc/system_control.h [hunk 2]
```diff
@@ -66,6 +67,7 @@ typedef enum USBRequestResult {
 typedef struct USBRequest {
   size_t size; // Structure size
   int type; // Request type (as defined by USBRequestType enum)
+  uint16_t value; // wValue field
```
Comment (sergeuz) [inline system/inc/system_control.h]: This change breaks ABI (new fields should be added at the end of the structure)

File: system/src/system_network_internal.h [hunk 3]
```diff
@@ -279,6 +279,7 @@ class ManagedNetworkInterface : public NetworkInterface
                 console.loop();
             }
 #if PLATFORM_THREADING
+            SystemISRTaskQueue.process();
```
Comment (sergeuz) [inline system/src/system_network_internal.h]: 👍

File: system/src/system_control.cpp [hunk 4]
```diff
@@ -292,15 +286,51 @@ void SystemControlInterface::setRequestResult(USBRequest* req, USBRequestResult
...
+    break;
+  }
+
+  case USB_REQUEST_DFU_MODE: {
+    System.dfu(false);
+    break;
+  }
+
+  case USB_REQUEST_LISTENING_MODE: {
+    setRequestResult(req, result);
```
Comment (sergeuz) [inline system/src/system_control.cpp]: Is there any problem with setting a result after calling `WiFi.listen()`? I just would like to see error handling unified in this function (if possible), i.e. make all case blocks to either call `setRequestResult()` and return, or assign to `result` variable.

File: system/src/system_control.cpp [hunk 5]
```diff
@@ -292,15 +286,51 @@ void SystemControlInterface::setRequestResult(USBRequest* req, USBRequestResult
...
+    break;
+  }
+
   default:
     if (usbReqAppHandler) {
       processAppRequest(data); // Forward request to the application thread
     } else {
       setRequestResult(req, USB_REQUEST_RESULT_ERROR);
     }
+    return;
```
Comment (sergeuz) [inline system/src/system_control.cpp]: Same here:
```cpp
if (usbReqAppHandler) {
  processAppRequest(data);
  return;
} else {
  result = USB_REQUEST_RESULT_ERROR;
}
```

**Comment by avtolstoy** [comment]
Thanks for the comments! Updated PR.

**Pull request merged**